langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/PadEnd.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.langlib.string;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BString;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.STRING_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.LENGTH_SHOULD_BE_LESS_THAN_2147483647;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Extern function ballerina.model.strings:padEnd.
+ *
+ * @since 2201.1.0
+ */
+
+public class PadEnd {
+
+    public static BString padEnd(BString str, long len, BString padChar) {
+        int strLength = str.length();
+        if (len <= strLength) {
+            return str;
+        }
+
+        if (len > Integer.MAX_VALUE) {
+            throw ErrorCreator.createError(getModulePrefixedReason(STRING_LANG_LIB,
+                            LENGTH_SHOULD_BE_LESS_THAN_2147483647));
+        }
```

Comment:
I think we do not need this check.
Ballerina `int` represents java `long`. So the user should be able to pass values greater than `Integer.MAX_VALUE`. If the user passes a value more than Long:MAX_VALUE it will be a compile error.

Reply:
I think you have done this because the length of the java string cannot exceed Integer:MAX_VALUE.

But by looking at ballerina's point of view use can pass a `len` greater than `Integer:MAX_VALUE`. So I suggest, if the `len` is greater than `Interger.MAX_VALUE`, set len = Interger.MAX_VALUE and continue without throwing an error.

@gimantha WDYT?

Reply:
+1

Reply:
on a second thought, if we ever thought of throwing an error for this java implementation in a future version, your suggestion can be a breaking change.

Reply:
Otherwise we can change the error into something like `length greater that '2147483647' not yet supported`?. We are already using these kinds of error msgs for implementation dependent errors. 
ex. 
```ballerina
int[int:MAX_VALUE] x = []; // array length greater that '2147483637' not yet supported
```